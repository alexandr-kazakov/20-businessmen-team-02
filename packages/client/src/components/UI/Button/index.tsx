import React, { ButtonHTMLAttributes, memo, FC } from 'react'
import classnames from 'classnames'
import { ButtonStyles } from './types'
import styles from './styles.module.scss'

type OwnProps = {
  variant: ButtonStyles
} & ButtonHTMLAttributes<HTMLButtonElement>

type Props = OwnProps

const Button: FC<Props> = props => {
  const { variant, children, ...buttonProps } = props

  const className = classnames(styles.button, {
    [styles.button_primary]: variant === ButtonStyles.primary,
    [styles.button_secondary]: variant === ButtonStyles.secondary,
  })

  return (
    <button {...buttonProps} className={className}>
      {children}
    </button>
  )
}

export default memo(Button)
