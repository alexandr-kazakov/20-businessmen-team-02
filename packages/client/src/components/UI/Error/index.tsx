import React from 'react'

import styles from './styles.module.scss'

interface ErrorBoundaryProps {
  children?: React.ReactNode
}

interface ErrorBoundaryState {
  error: Error | null
}

class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props)
    this.state = { error: null }
  }

  static getDerivedStateFromError(error: Error) {
    return { error }
  }

  render() {
    const { error } = this.state

    if (error) {
      return (
        <div className={styles.container}>
          <div className={styles.title}>Что то пошло не так... ( ˘︹˘ )</div>

          {error.message && (
            <div className={styles.error}>Текст ошибки: {error.message}</div>
          )}
        </div>
      )
    }

    return this.props.children
  }
}

export default ErrorBoundary
