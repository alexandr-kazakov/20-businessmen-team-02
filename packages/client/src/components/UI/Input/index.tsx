import React, { type InputHTMLAttributes, memo } from 'react'
import classnames from 'classnames'

import styles from './styles.module.scss'

type InputProps = {
  isValid?: boolean
} & InputHTMLAttributes<HTMLInputElement>

export const Input: React.FC<InputProps> = memo(({ isValid = true, ...props }) => {
  const className = classnames(styles.input, props.className, {
    [styles.error]: !isValid,
  })

  return <input {...props} className={className} />
})
