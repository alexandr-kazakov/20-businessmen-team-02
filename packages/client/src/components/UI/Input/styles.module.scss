.input {
  border: none;
  padding: 10px;
  border-radius: 10px;
  font-style: normal;
  font-weight: 500;
  font-size: 13px;
  line-height: 13px;
  color: #1e1e1e;

  &.error {
    background-color: rgb(204, 85, 85);
  }
}
