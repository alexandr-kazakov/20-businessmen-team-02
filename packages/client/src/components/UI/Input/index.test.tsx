import React from 'react'
import { render } from '@testing-library/react'
import {Input} from '@/components/UI/Input'

test('should render Input component', () => {
  const {asFragment} = render(<Input />)

  expect(asFragment()).toMatchSnapshot()
})