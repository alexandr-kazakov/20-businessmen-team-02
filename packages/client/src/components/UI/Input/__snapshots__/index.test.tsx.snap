// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input component 1`] = `
<DocumentFragment>
  <input
    class=""
  />
</DocumentFragment>
`;
